src/view-resources.js

```diff
@@ -67,6 +73,72 @@ interface ViewEngineHooks {
 * Represents a collection of resources used during the compilation of a view.
 */
 export class ViewResources {
+
+  static convention(target: Function) {
+    let config = target.resource;
+    let resource;
+    if (config) {
+      if (typeof config === 'string') {
+        // it's a custom element, with name is the resource variable
+        // static resource = 'my-element'
+        resource = new HtmlBehaviorResource();
+        resource.elementName = config;
+      } else {
+        if (typeof config === 'function') {
+          // static resource() {  }
+          config = config();
+        }
+        if (typeof config === 'string') {
+          // static resource() { return 'my-custom-element-name' }
+          config = { name: config };
+        }
+        // after normalization, copy to another obj
+        config = Object.assign({}, config);
+        // no type specified = custom element
+        let resourceType = config.type || 'element';
+        switch (resourceType) {
+          case 'element': case 'attribute':
+            resource = new HtmlBehaviorResource();
+            if (resourceType === 'element') {
+              resource.elementName = _hyphenate(config.name || target.name);
+            } else {
+              resource.attributeName = _hyphenate(config.name || target.name);
+            }
+            if (config.templateController) {
```

Comment:
Nit: I feel like `in` would be more consistent with `defaultBindingMode` check just below.
It would also avoid me thinking about whether ignoring `templateController: false` is totally ok or has tiny chance of screwing us up.